src/klein/test/test_headers_compat.py

```diff
@@ -30,12 +30,13 @@ def _twistedHeaderNormalize(value):
     Normalize the given header value according to the rules of the installed
     Twisted version.
     """
-    if _sanitizeLinearWhitespace is not None:
-        valueBytes = value.encode("utf-8")
-        valueBytes = _sanitizeLinearWhitespace(valueBytes)
-        value = valueBytes.decode("utf-8")
-
-    return value
+    if _sanitizeLinearWhitespace is None:
+        return value
+    else:
+        return cast(
+            Text,
+            _sanitizeLinearWhitespace(value.encode("utf-8")).decode("utf-8"),
```

Comment:
It looks like https://github.com/twisted/twisted/pull/1284 would be helpful here.

Reply:
It dawns on me that the `mypy` Tox environment… doesn't pin Twisted, so we should see some "cast not needed" errors after Twisted ships with type hints in code used by Klein…

I'll have a think on that for another PR.